src/main/java/org/folio/circulation/resources/LoanCollectionResourceHelper.java

```diff
@@ -0,0 +1,112 @@
+package org.folio.circulation.resources;
+
+import org.folio.circulation.domain.Item;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.RequestQueue;
+import org.folio.circulation.domain.ServicePointRepository;
+import org.folio.circulation.domain.User;
+import org.folio.circulation.domain.validation.ItemNotFoundValidator;
+import org.folio.circulation.support.Result;
+
+import java.util.concurrent.CompletableFuture;
+
+import static org.folio.circulation.domain.representations.LoanProperties.ITEM_ID;
+import static org.folio.circulation.support.Result.of;
+import static org.folio.circulation.support.ValidationErrorFailure.failedValidation;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+public class LoanCollectionResourceHelper {
+
+  private LoanCollectionResourceHelper(){
+
+  }
+
+  public static Result<LoanAndRelatedRecords> addItem(
+    Result<LoanAndRelatedRecords> loanResult,
+    Result<Item> item) {
+
+    return Result.combine(loanResult, item,
+      LoanAndRelatedRecords::withItem);
+  }
+
+  public static Result<LoanAndRelatedRecords> addRequestQueue(
+    Result<LoanAndRelatedRecords> loanResult,
+    Result<RequestQueue> requestQueueResult) {
+
+    return Result.combine(loanResult, requestQueueResult,
+      LoanAndRelatedRecords::withRequestQueue);
+  }
+
+  public static Result<LoanAndRelatedRecords> addUser(
+    Result<LoanAndRelatedRecords> loanResult,
+    Result<User> getUserResult) {
+
+    return Result.combine(loanResult, getUserResult,
+      LoanAndRelatedRecords::withRequestingUser);
+  }
+
+  static Result<LoanAndRelatedRecords> refuseWhenHoldingDoesNotExist(
```

Comment:
We usually use validators for such cases, see `org.folio.circulation.domain.validation` package.

Reply:
@bohdan-suprun I just want to add that this method was just replaced from one class to other. There aren`t any changes from the previous version in this implementation.

Reply:
Refactored.